tests/foreman/api/test_parameters.py

```diff
@@ -51,6 +50,7 @@ def test_positive_parameter_precedence_impact(
         group_parameters_attributes=[{'name': param_name, 'value': param_value}],
     ).create()
     request.addfinalizer(hg.delete)
+    request.addfinalizer(host.delete)
```

Comment:
I think here host should be deleted before hg, as host is linked to hg.
And, Here instead of 2-request.addfinalizer's, I'd suggest using a finalizer method like in the original PR for this test, which is used work correctly for teardown
https://github.com/SatelliteQE/robottelo/pull/13620

Reply:
@Gauravtalreja1 
To your point about the order of deletion:
```
Finalizers are executed in a first-in-last-out order.
```
https://docs.pytest.org/en/latest/how-to/fixtures.html#note-on-finalizer-order

What is your motivation behind the suggestion to use a single finalizer method?

Reply:
Just to simplify this, Its better to write a single method with the execution order we provide, instead of pytest to decide this order for finalizers

Reply:
What about the reasons for changing it to finalizers in https://github.com/SatelliteQE/robottelo/pull/14036 ?

>  so basically when some assertion defined in the test before the finalizer part fails, finalizer is not added and so not executed.

This is also why I first wanted to go for the fixture approach.